File: lean_d13critical/VerifyBridge.java
import java.math.BigInteger;

/** Independent exact checker for the Round-25 bridge equations. */
public final class VerifyBridge {
  private static final BigInteger TEN = BigInteger.TEN;
  private static final BigInteger ONE = BigInteger.ONE;

  private static void check(int p, int e, int digits, int d, int c) {
    BigInteger P = BigInteger.valueOf(p);
    BigInteger D = BigInteger.valueOf(d);
    BigInteger C = BigInteger.valueOf(c);
    BigInteger numerator = C.multiply(TEN.pow(digits)).add(ONE);
    BigInteger[] qr = numerator.divideAndRemainder(D);
    if (qr[1].signum() != 0) throw new AssertionError("nonintegral quotient");
    BigInteger q = qr[0];
    if (!P.pow(e).subtract(TEN).equals(D)) throw new AssertionError("wrong D");
    if (!BigInteger.TEN.multiply(BigInteger.TEN.multiply(P).add(BigInteger.valueOf(e))).equals(C))
      throw new AssertionError("wrong C");
    if (q.toString().length() != digits) throw new AssertionError("wrong digits");
    if (!D.multiply(q).equals(numerator)) throw new AssertionError("bridge equation");

    BigInteger wrongDigits = C.multiply(TEN.pow(digits - 1)).add(ONE);
    if (D.multiply(q).equals(wrongDigits)) throw new AssertionError("digit mutation accepted");
    int reversed = 10 * (10 * e + p);
    BigInteger wrongOrder = BigInteger.valueOf(reversed).multiply(TEN.pow(digits)).add(ONE);
    if (D.multiply(q).equals(wrongOrder)) throw new AssertionError("order mutation accepted");
  }

  public static void main(String[] args) {
    check(7, 4, 733, 2391, 740);
    check(3, 7, 666, 2177, 370);
    System.out.println("JAVA_BRIDGE_CHECK_PASS cases=2 mutations=4");
  }
}

File: lean_d13critical/verify_bridge.py
#!/usr/bin/env python3
"""Independent exact checker for the Round-25 bridge equations."""
import json
from pathlib import Path

ROOT = Path(__file__).resolve().parent
cert = json.loads((ROOT / "certificate.json").read_text())

cases = [
    ("q1_3", 7, 4, 733, 2391, 740),
    ("q2_0", 3, 7, 666, 2177, 370),
]

for name, p, e, digits, D, C in cases:
    q = int(cert[name]["n"])
    assert p**e - 10 == D
    assert 10 * (10 * p + e) == C
    assert len(str(q)) == digits
    assert D * q == C * 10**digits + 1
    # The linear equation has at most one natural-number solution.
    assert q == (C * 10**digits + 1) // D

# Semantic mutations: wrong digit stratum and reversed prime/exponent block.
q1 = int(cert["q1_3"]["n"])
q2 = int(cert["q2_0"]["n"])
assert 2391 * q1 != 740 * 10**732 + 1
assert 2177 * q2 != 370 * 10**665 + 1
assert 2391 * q1 != 470 * 10**733 + 1
assert 2177 * q2 != 730 * 10**666 + 1

print("PYTHON_BRIDGE_CHECK_PASS cases=2 mutations=4")
